pyspinw/dimensionality.py

```diff
@@ -0,0 +1,129 @@
+""" Tools for checking input shapes """
```

Comment:
I think I suggested checking vector/array sizes - I figured there might be a class of bugs where people converting their MATLAB scripts and not taking care of memory order etc..
It's nice to know this can now be done with type hinting, but I believe this doesn't do any runtime check, is that right? Maybe pydantic can do it?

Reply:
Ah right... sorry yes, type hinting is only for static checking, it doesn't work at runtime.